Pear/Profile/EditProfileSectionsViewController.swift

```diff
@@ -145,3 +230,66 @@ extension EditProfileSectionsViewController: UITableViewDataSource {
 extension EditProfileSectionsViewController: UITableViewDelegate {
     
 }
+
+extension EditProfileSectionsViewController: didUpdateInterestsDelegate, didUpdateGroupsDelegate, didUpdatePromptsDelegate {
+    func updateInterests(updatedUser: UserV2, newInterests: [Interest]) {
+        /// Update the global "interests" variable with the newly edited [Interest] that was passed in from the InterestSettingsVC + newly updated User
+        self.interests = newInterests
+        self.updatingUser = updatedUser
+        
+        /// Network call to update the user's interest on the backend
+        let selectedInterestsIds = self.interests.map { $0.id }
+        NetworkManager.updateInterests(interests: selectedInterestsIds) { [weak self] (success) in
+            guard let self = self else { return }
+            DispatchQueue.main.async {
+                if success {
+                } else {
+                    self.present(UIAlertController.getStandardErrortAlert(), animated: true, completion: nil)
+                }
+            }
+        }
+        
+        /// Passing the updatedUser back to the EditProfileViewController to reflect the change in this session
+        delegate?.updateUser(updatedUser: updatingUser)
+        self.fadeTableView.view.reloadData()
+    }
+    
+    func updateGroups(updatedUser: UserV2, newGroups: [Group]) {
+        self.groups = newGroups
+        self.updatingUser = updatedUser
+        
+        /// Network call to update the user's groups on the backend
+        let selectedGroupsIds = self.groups.map { $0.id }
+        NetworkManager.updateGroups(groups: selectedGroupsIds) { [weak self] (success) in
+            guard let self = self else { return }
+            DispatchQueue.main.async {
+                if success {
+                } else {
+                    self.present(UIAlertController.getStandardErrortAlert(), animated: true, completion: nil)
+                }
+            }
+        }
+        delegate?.updateUser(updatedUser: updatingUser)
+        self.fadeTableView.view.reloadData()
+    }
+    
+    func updatePrompts(updatedUser: UserV2, newPrompt: Prompt) {
+        // Set global variable prompt
+        self.prompts = updatedUser.prompts
+        
+        // Set the user in this to this passed in user
+        self.updatingUser = updatedUser
+        delegate?.updateUser(updatedUser: self.updatingUser)
+        
+    }
+}
+
+/// Used for when the user is adding a new prompt and creating a new answer for it, rather than editing an existing one.
+extension EditProfileSectionsViewController: didCreateNewPrompt {
+    func addPrompt(newPrompt: Prompt) {
```

Comment:
Add newline before `addPrompt`

You may also want to include this delegate function inside that single protocol I was talking about